src/scaffoldmaker/meshtypes/meshtype_3d_bladder1.py

```diff
@@ -880,13 +573,676 @@ def defineFaceAnnotations(cls, region, options, annotationGroups):
+
+def findNodesAlongBladderNeck(sx_dome_group, sx_neck_group, domeSegmentLength, neckSegmentLength, elementsCountAround, elementsCountAlongNeck):
+
+    # Transition
+    transitLength = (domeSegmentLength + neckSegmentLength) / 2
+    addingLength = transitLength - neckSegmentLength
+    if transitLength <= neckSegmentLength:
+        e = 1
+        xi = transitLength / neckSegmentLength
+    else:
+        e = int(transitLength / neckSegmentLength) + 1
+        xi = transitLength / (e * neckSegmentLength)
+    xTransition = interp.interpolateCubicHermite(sx_dome_group[0][-1], sx_dome_group[1][-1], sx_neck_group[0][e], sx_neck_group[1][e], xi)
+    # d1Transition = interp.interpolateCubicHermiteDerivative(sx_dome_group[0][-1], sx_dome_group[1][-1], sx_neck_group[0][e], sx_neck_group[1][e], xi)
+    d2Transition = interp.interpolateCubicHermite(sx_dome_group[2][-1], sx_dome_group[3][-1], sx_neck_group[2][e], sx_neck_group[3][e], xi)
+    d3Transition = interp.interpolateCubicHermite(sx_dome_group[4][-1], sx_dome_group[5][-1], sx_neck_group[4][e], sx_neck_group[5][e], xi)
+    px_transit, pd1_transit = createEllipsePoints(xTransition, 2 * math.pi, d2Transition, d3Transition, elementsCountAround,
+                                                  startRadians=0.0)
+
+    # Create ellipses along neck around the central path
+    xEllipses_neck = []
+    d1Ellipses_neck = []
+    for n in range(0, len(sx_neck_group[0])):
+        px, pd1 = createEllipsePoints(sx_neck_group[0][n], 2 * math.pi, sx_neck_group[2][n], sx_neck_group[4][n], elementsCountAround,
+                                      startRadians=0.0)
+        xEllipses_neck.append(px)
+        d1Ellipses_neck.append(pd1)
+
+    # Find d2
+    d2Raw = []
+    for n1 in range(elementsCountAround):
+        xAlong = []
+        d2Along = []
+        for n2 in range(len(xEllipses_neck) - 1):
+            v1 = xEllipses_neck[n2][n1]
+            v2 = xEllipses_neck[n2 + 1][n1]
+            d2 = findDerivativeBetweenPoints(v1, v2)
+            xAlong.append(v1)
+            d2Along.append(d2)
+        xAlong.append(xEllipses_neck[-1][n1])
+        d2Along.append(d2)
+        d2Smoothed = interp.smoothCubicHermiteDerivativesLine(xAlong, d2Along)
+        d2Raw.append(d2Along)
+
+    # Rearrange d2
+    d2Ellipses_neck = []
+    for n2 in range(len(xEllipses_neck)):
+        d2Around = []
+        for n1 in range(elementsCountAround):
+            d2 = d2Raw[n1][n2]
+            d2Around.append(d2)
+        d2Ellipses_neck.append(d2Around)
+
+    # Spread out elements along neck of the bladder
+    xRawNeck = []
+    d2RawNeck = []
+    for n1 in range(elementsCountAround):
+        xAlong = []
+        d2Along = []
+        for n2 in range(len(xEllipses_neck)):
+            xAlong.append(xEllipses_neck[n2][n1])
+            d2Along.append(d2Ellipses_neck[n2][n1])
+        xSampledAlongNeck, d2SampledAlongNeck = interp.sampleCubicHermiteCurves(xAlong, d2Along,
```

Comment:
sampleCubicHermiteCurvesSmooth may work better here; need to specify starting derivative to match body, and end derivative could be prescribed for a certain element density along the urethra, or left free to just smoothly adapt.

Reply:
Since the addLengthStart was added here to include the transition between dome and neck, the function "sampleCubicHermiteCurves" has been used.